source/extensions/filters/network/redis_proxy/redirection_mgr_impl.cc

```diff
@@ -0,0 +1,60 @@
+#include "extensions/filters/network/redis_proxy/redirection_mgr_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RedisProxy {
+
+bool RedirectionManagerImpl::onRedirection(const std::string& cluster_name) {
+  auto it = info_map_.find(cluster_name);
+  if (it != info_map_.end()) {
+    ClusterInfoSharedPtr info = it->second;
```

Comment:
is there a race here between onRedirection and unregisterCluster?

Reply:
unregisterCluster is called only on the main thread, and onRedirection is called by the worker threads. I suppose it is possible for a race, if a cluster is destroyed without all network filters that are referencing that cluster being shutdown/destroyed first. Is that possible?

Reply:
I think that's possible.  The cluster can be removed/reloaded in the main thread first, and that triggers the clean up for the network filters.

Reply:
ok ... I'll rework